app/assets/javascripts/spotlight/blocks/block.js

```diff
@@ -1,132 +1,16 @@
-        var key = $(input).data('key') || input.getAttribute('name');
-        if (key) {
-          data[key] = null;
-        }
-      });
-
-      if(inputs.length > 0) {
-        inputs.each(function(index,input){
-          var key = $(input).data('key') || input.getAttribute('name');
-          if (key) {
-            if($(input).is(':checkbox') && $(input).val() == "true") {
-              data[key] = true;
-            } else {
-              data[key] = $(input).val();
-            }
-          }
-        });
-      }
-
-      return data;
-    },
-
-    loadFormDataByKey: function(data) {
-      this.$(':input').not('button').each(function(index, input) {
-        var key = $(input).data('key') || input.getAttribute('name');
-        // by wrapping it in an array, it'll "just work" for radio and checkbox fields too
-        $(this).val([data[key]]);
-      });
-    },
-
-    loadData: function(data){
-      if (this.hasTextBlock()) {
-        this.getTextBlock().html(SirTrevor.toHTML(data.text, this.type));
-      }
-      this.loadFormDataByKey(data);
-      this.afterLoadData(data);
-    },
-
-    afterLoadData: function(data) { },
-
-    caption_field_template: ['<option value="<%= field %>"><%= label %></option>'].join("\n"),
-
-    loadCaptionField: function(){
-      var block = this;
-      var metadata_url = $('form[data-metadata-url]').data('metadata-url');
-      var primary_caption_field = $('#' + this.formId(this.primary_field_key));
-      var secondary_caption_field = $('#' + this.formId(this.secondary_field_key));
-      var primary_caption_selected_value = primary_caption_field.data("select-after-ajax");
-      var secondary_caption_selected_value = secondary_caption_field.data("select-after-ajax");
-      $.ajax({
-        accepts: "json",
-        url: metadata_url
-      }).success(function(data){
-        // Only checking the primary caption field
-        // Could check both but I'm not sure it's necessary
-        if($("option", primary_caption_field).length == 2){
-          var options = "";
-          $.each(data, function(i, field){
-            options += _.template(block.caption_field_template)(field);
-          });
-
-          primary_caption_field.append(options);
-          secondary_caption_field.append(options);
-
-          primary_caption_field.val([primary_caption_selected_value]);
-          secondary_caption_field.val([secondary_caption_selected_value]);
-          // re-serialize the form so the form observer
-          // knows about the new drop down options.
-          serializeFormStatus($('form[data-metadata-url]'));
+    beforeBlockRender: function() {
+      this.availableMixins.forEach(function(mixin) {
+        if (this[mixin] && SirTrevor.BlockMixins[_.capitalize(mixin)].preload) {
+          this.withMixin(SirTrevor.BlockMixins[_.capitalize(mixin)]);
         }
-      });
+      }, this);
     },
-
-    addCaptionSelectFocus: function(){
```

Comment:
I actually couldn't get the old behavior to work for me at all, but I have connected it so when you make a selection, it will check the checkbox for you. I left the markup for what I think is the equivalent to `addCaptionSelectFocus`, so it you want to give it a go..